src/jupyter_contrib_nbextensions/nbextensions/code_prettify/code_prettify.js

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Jupyter-Contrib Team.
+// Distributed under the terms of the Modified BSD License.
+
+define(function(require, exports, module) {
+    'use strict';
+
+    var Jupyter = require('base/js/namespace');
+    var keyboard = require('base/js/keyboard');
+    var utils = require('base/js/utils');
+    var configmod = require('services/config');
+    var Cell = require('notebook/js/cell').Cell;
+    var CodeCell = require('notebook/js/codecell').CodeCell;
+
+    var add_edit_shortcuts = {};
+    var replace_in_cell = false; //bool to enable/disable replacements 
+    var exec_code_verbose = true;
+    var cfg = {
+        code_format_hotkey: 'Ctrl-L',
+    }
+
+    var kMap = { // map of parameters for supported kernels
+        python2: {
+            library: 'from yapf.yapflib.yapf_api import FormatCode',
+            exec: yapf_format,
+            post_exec: ''
+        },
+        python3: {
+            library: 'from yapf.yapflib.yapf_api import FormatCode',
+            exec: yapf_format,
+            post_exec: ''
+        },
+        ir: {
```

Comment:
ok  indeed, `ir` is the default name but users can specify a [different name at install](https://github.com/IRkernel/IRkernel). There is no language tag accessible in the kernel object, so no indication of what is the underlying language. 
I have added a map connecting names with the associated language and use the language as the key for selecting the prettifier.  For  other names defined by users, I am afraid that they have to be added by hand (or perhaps via some parameters in the yaml?)